src/tiledimage.js

```diff
@@ -1350,32 +1350,23 @@ $.extend($.TiledImage.prototype, $.EventSource.prototype, /** @lends OpenSeadrag
             );
             if(tile && tile.isBottomMost && tile.isRightMost && tile.loaded){
                 levelList.push(level);
-                levelList.hasHigherResolutionFallback = true;
                 break;
             }
         }
 
 
-        // Update any level that will be drawn
+        // Update any level that will be drawn.
+        // We are iterating from highest resolution to lowest resolution
+        // Once a level fully covers the viewport the loop is halted and
+        // lower-resolution levels are skipped
         for (let i = 0; i < levelList.length; i++) {
             let level = levelList[i];
-            var drawLevel = false;
 
-            //Avoid calculations for draw if we have already drawn this
             var currentRenderPixelRatio = this.viewport.deltaPixelsFromPointsNoRotate(
                 this.source.getPixelRatio(level),
                 true
             ).x * this._scaleSpring.current.value;
 
-            if (i === levelList.length - 1 ||
```

Comment:
I see. Presumably you ran some tests and observed this? Is there a chance it would not be the case in some circumstances? It doesn't seem like it hurts anything to have the guard here just in case.

Reply:
> I see. Presumably you ran some tests and observed this? Is there a chance it would not be the case in some circumstances? It doesn't seem like it hurts anything to have the guard here just in case.

So I did do some testing and I did observe this. But I also looked at how the calculation is done within `getLevelsInterval`. I think that division by `this.minPixelRatio` within the `Math.log`, followed by `Math.floor` which rounds down, guarantees that `currentRenderPixelRatio >= this.minPixelRatio` is true for all the levels we're iterating over - because if it weren't, the ratio inside the `Math.log` call would fall below 1, causing `Math.log` to be negative, and we'd round down to the next lower level as the highest one to start with.

So I think it is mathematically defined that this guard is never going to be triggered.

Reply:
I'm afraid it's too complicated for me to confidently say it would never happen. I guess I'm wondering what benefit we get from removing this clause... If it's just some redundant safety, maybe it's worth keeping in? Or are you thinking it's actually causing problems?

Reply:
I don't think it was causing any problems with code execution (since I don't believe it was doing anything at all). The reason I wanted to remove it was to simplify the code so it is more understandable, since more understandable is more maintainable. `_getLevelsInterval` is only used in one place, which is to calculated the levels to use in this loop. If that were incorrect, I think ideally it should be fixed there so that function gave accurate levels, rather than second-guessing that calculation during the loop itself. (Note: I don't think that function is incorrect or I would have tried to fix it).

Regardless, I've added the guard back in. There are slightly different variables names and it is a little more streamlined, but it does the same thing it did before.